llvm/lib/Target/AMDGPU/SIMemoryLegalizer.cpp

```diff
@@ -2802,7 +2801,10 @@ bool SIMemoryLegalizer::expandAtomicCmpxchgOrRmw(const SIMemOpInfo &MOI,
 bool SIMemoryLegalizer::runOnMachineFunction(MachineFunction &MF) {
   bool Changed = false;
 
-  SIMemOpAccess MOA(MF);
+  const MachineModuleInfo &MMI =
+      getAnalysis<MachineModuleInfoWrapperPass>().getMMI();
```

Comment:
Need to add this to the `AnalysisUsage`?

Reply:
It's already in the base MachineFunction getAnalysisUsage